src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -49,6 +53,8 @@ private ObjectMethodExecutor(MethodInfo methodInfo)
 
         public MethodInfo MethodInfo { get; }
 
+        public ParameterInfo[] ActionParameters { get; }
```

Comment:
MethodInfo.GetParameters() does allocation of ParameterInfo[] for every invocation. In order to avoid repeated allocations of the array for every request, this property is made available in ObjectMethodExecutor that can be used in PrepareArguments.